test/CadlRanchProjects/type/property/nullable/src/Generated/Models/CollectionsByteProperty.cs

```diff
@@ -18,14 +18,13 @@ public partial class CollectionsByteProperty
         /// <summary> Initializes a new instance of CollectionsByteProperty. </summary>
         /// <param name="requiredProperty"> Required property. </param>
         /// <param name="nullableProperty"> Property. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="requiredProperty"/> or <paramref name="nullableProperty"/> is null. </exception>
+        /// <exception cref="ArgumentNullException"> <paramref name="requiredProperty"/> is null. </exception>
         internal CollectionsByteProperty(string requiredProperty, IEnumerable<BinaryData> nullableProperty)
         {
             Argument.AssertNotNull(requiredProperty, nameof(requiredProperty));
-            Argument.AssertNotNull(nullableProperty, nameof(nullableProperty));
 
             RequiredProperty = requiredProperty;
-            NullableProperty = nullableProperty.ToList();
+            NullableProperty = nullableProperty?.ToList();
```

Comment:
This needs to be initialized as a ChangeTrackingList if its `nullable` or `optional`.  If we use Optional<T> here the .ToList takes care of that for us.

Reply:
I agree that we should initialize this as a ChangeTrackingList and never have a list to be null anywhere
But here we cannot use Optional.ToList because this is in the public constructor. The `Optional<T>` only appear in the deserialization method.